src/hb-mutex.hh

```diff
@@ -81,20 +81,26 @@ typedef int hb_mutex_impl_t;
 
 #else
 
-#error "Could not find any system to define mutex macros."
-#error "Check hb-mutex.hh for possible resolutions."
+#include <mutex>
+#include <new>
+typedef std::mutex              hb_mutex_impl_t;
```

Comment:
After ensuring that this is actually being used everywhere and building, it appears that most std::mutex implementations are going to want to link against the c++ standard library (mostly for exception types apparently), so may need to have a feature test for that?